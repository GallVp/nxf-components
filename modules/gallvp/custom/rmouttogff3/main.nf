process CUSTOM_RMOUTTOGFF3 {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/perl-bioperl:1.7.8--hdfd78af_1':
        'biocontainers/perl-bioperl:1.7.8--hdfd78af_1' }"

    input:
    tuple val(meta), path(rmout)

    output:
    tuple val(meta), path("*.gff3") , emit: gff3
    path "versions.yml"             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    shell:
    prefix = task.ext.prefix ?: "${meta.id}"
    template 'rmouttogff3.pl'

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.gff3

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        perl: \$(perl --version | sed -n 's|This is perl.*(v\\(.*\\)) .*|\\1|p'  )
    END_VERSIONS
    """
}
