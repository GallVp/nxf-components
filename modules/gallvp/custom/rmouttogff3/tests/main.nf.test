nextflow_process {

    name "Test Process CUSTOM_RMOUTTOGFF3"
    script "../main.nf"
    process "CUSTOM_RMOUTTOGFF3"

    tag "modules"
    tag "modules_gallvp"
    tag "custom"
    tag "custom/rmouttogff3"
    tag "repeatmasker/repeatmasker"

    setup {
        run("REPEATMASKER_REPEATMASKER") {
            script "../../../repeatmasker/repeatmasker/main.nf"
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = []
                """
            }
        }
    }

    test("sarscov2 - genome - fasta - repeatmasker - out") {

        when {
            process {
                """
                input[0] = REPEATMASKER_REPEATMASKER.out.out
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - genome - fasta - repeatmasker - out - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = input[0] = REPEATMASKER_REPEATMASKER.out.out
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}