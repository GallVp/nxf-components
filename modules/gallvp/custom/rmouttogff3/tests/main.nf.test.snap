{
    "sarscov2 - genome - fasta - repeatmasker - out - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gff3:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,46901143ed4508b93cb4b64cd0b352f2"
                ],
                "gff3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gff3:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,46901143ed4508b93cb4b64cd0b352f2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-27T16:27:50.770327"
    },
    "sarscov2 - genome - fasta - repeatmasker - out": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gff3:md5,689df952b225e56b521d12f0dfab2ab8"
                    ]
                ],
                "1": [
                    "versions.yml:md5,029655f0760e918db2ef104e09d379c1"
                ],
                "gff3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gff3:md5,689df952b225e56b521d12f0dfab2ab8"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,029655f0760e918db2ef104e09d379c1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-27T16:49:03.738128"
    }
}